curriculum/challenges/english/20-upcoming-python/learn-data-structures-by-building-the-merge-sort-algorithm/6569c2cbf6c993ea8cd85682.md

```diff
@@ -0,0 +1,76 @@
+---
+id: 6569c2cbf6c993ea8cd85682
+title: Step 31
+challengeType: 20
+dashedName: step-31
+---
+
+# --description--
+
+At this point, the `numbers` list has been sorted. Let's print the sorted list.
+
+Start by creating a `print()` function that prints string `Sorted array: ` and the the sorted list.
+
+You can do this by concatenating `"Sorted array: "` and `str(numbers)` in the `print` function.
+
+The `str` function converts the list to a string.
+
+# --hints--
+
+You should have `print("sorted array: " + str(numbers))`. Remember to add the space after the colon
+
+```js
+assert.match(code, /print\("sorted numbers: " \+ str\(\s*numbers\s*\)\)/)
+```
+
+# --seed--
+
+## --seed-contents--
+
+```py
+def merge_sort(array):
+    if len(array) <= 1:
+        return
+    
+    middle_point = len(array) // 2
+    left_part = array[:middle_point]
+    right_part = array[middle_point:]
+
+    merge_sort(left_part)
+    merge_sort(right_part)
+
+    left_array_index = 0
+    right_array_index = 0
+    sorted_index = 0
+
+    while left_array_index < len(left_part) and right_array_index < len(right_part):
+        if left_part[left_array_index] < right_part[right_array_index]:
+            array[sorted_index] = left_part[left_array_index]
+            left_array_index += 1
+        else:
+            array[sorted_index] = right_part[right_array_index]
+            right_array_index += 1
+        sorted_index += 1
+
+    while left_array_index < len(left_part):
+        array[sorted_index] = left_part[left_array_index]
+        left_array_index += 1
+        sorted_index += 1
+    
+    while right_array_index < len(right_part):
+        array[sorted_index] = right_part[right_array_index]
+        right_array_index += 1
+        sorted_index += 1
+
+if __name__ == '__main__':
+    numbers = [4, 10, 6, 14, 2, 1, 8, 5]
+
+    print("Unsorted array: ")
+    print(numbers)
+
+    merge_sort(numbers)
+
+--fcc-editable-region--
+
+--fcc-editable-region--
+```
```

Comment:
Please add the solution. See this as a reference: https://github.com/freeCodeCamp/freeCodeCamp/blob/4a365edfa6890afbdaacb2bc3b9ac38a0ed418c8/curriculum/challenges/english/20-upcoming-python/learn-recursion-by-solving-the-tower-of-hanoi-puzzle/657b6a4a49faec5c600287ba.md?plain=1#L58-L84